wrapanapi/systems/openstack.py

```diff
@@ -938,28 +924,25 @@ def with_volumes(self, *configurations, **kwargs):
             raise ValueError("n does not equal the length of configurations")
         # now n == len(configurations)
         volumes = []
-        try:
-            for configuration in configurations:
-                if isinstance(configuration, int):
-                    size, kwargs = configuration, {}
-                elif len(configuration) == 1:
-                    size, kwargs = configuration[0], {}
-                elif len(configuration) == 2:
-                    size, kwargs = configuration
-                else:
-                    size = configuration[0]
-                    kwargs = configuration[1]
-                volumes.append(self.create_volume(size, **kwargs))
-            yield volumes
-        finally:
-            self.delete_volume(*volumes)
+        for configuration in configurations:
+            if isinstance(configuration, int):
+                size, kwargs = configuration, {}
+            elif len(configuration) == 1:
+                size, kwargs = configuration[0], {}
+            elif len(configuration) == 2:
+                size, kwargs = configuration
+            else:
+                size = configuration[0]
+                kwargs = configuration[1]
+            volumes.append(self.create_volume(size, **kwargs))
+        return volumes
 
     def volume_attachments(self, volume_id):
         """Returns a dictionary of ``{instance: device}`` relationship of the volume."""
         volume = self.capi.volumes.get(volume_id)
         result = {}
         for attachment in volume.attachments:
-            result[self.get_vm(attachment['server_id']).name] = attachment['device']
+            result[self.get_vm(id=attachment['server_id']).name] = attachment['device']
```

Comment:
I think readability would be a bit better here splitting the get_vm call to a separate line. Totally optional though.

Reply:
maybe it would be nicer to use list comprehension. Let's see how it looks like...